### PR TITLE
Use hitTest for sampler overlay highlighting

### PR DESCRIPTION
**Pull request by volfied**
Adresses #3111 - we were using our own bounds calculations, which would consider masked layers in their full rectangle. Now we rely on hitTest, similar to what we do with superselect, so masked out areas don't highlight the layer.

### CODE REVIEW

**Comment by iwehrman** [comment]
It worries me that the promise returned from hitTest isn't used to synchronize this function. It seems like we could be executing a lot of these update calls concurrently. Should this function return a promise and then be throttled?

File: src/js/jsx/tools/SamplerOverlay.jsx [hunk 1]
```diff
@@ -518,34 +520,49 @@ define(function (require, exports, module) {
...
-                    return;
-                }
+            uiUtil.hitTestLayers(this.state.document.id, canvasMouse.x, canvasMouse.y)
+                .bind(this)
+                .then(function (hitLayerIDs) {
+                    var selectableLayers = this.state.document.layers.selectable,
+                        selectableLayerIDs = collection.pluck(selectableLayers, "id"),
+                        considerIDs = collection.intersection(hitLayerIDs, selectableLayerIDs);
+
+                    return considerIDs.findLast(function (id) {
```
Comment (pineapplespatula) [inline src/js/jsx/tools/SamplerOverlay.jsx]: what happens in the case where there isn't a layer returned by hitTestLayers?

**Comment by volfied** [comment]
@iwehrman we do the same thing in superselect highlighting as well. If we throttle/debounce this, then highlighting will appear broken to the user and look worse. So far it worked ok, and I'm hoping with the proposed hitTest improvements on the core side, things will work faster in the future.

@pineapplespatula when hitTest returns an empty array, intersection is also empty, and nothing gets highlighted.

**Comment by mcilroyc** [comment]
nudging to @iwehrman because he had a concern that may be explained by barkin's comment.

**Comment by iwehrman** [comment]
I don't love it, but OK.

**Pull request merged**